### PR TITLE
follow up #18517

### PR DESCRIPTION
**Pull request by ringabout**
Or I should remove the line of `parentDir`?

### CODE REVIEW

File: lib/pure/os.nim [hunk 1]
```diff
@@ -11,10 +11,13 @@
...
 
-runnableExamples("-r:off"):
+runnableExamples:
   let myFile = "/path/to/my/file.nim"
   assert splitPath(myFile) == (head: "/path/to/my", tail: "file.nim")
-  assert parentDir(myFile) == "/path/to/my"
+  when defined(posix):
+    assert parentDir(myFile) == "/path/to/my"
+  else:
+    assert parentDir(myFile) == "\\path\\to\\my"
```
Comment (timotheecour) [inline lib/pure/os.nim]:
```suggestion
  assert parentDir(myFile) == "/path/to/my".unixToNativePath
```
try this

**Review by timotheecour — APPROVED** [review]

File: lib/pure/os.nim [hunk 2]
```diff
@@ -11,10 +11,10 @@
...
 ## working with directories, running shell commands, etc.
 
-runnableExamples("-r:off"):
+runnableExamples:
   let myFile = "/path/to/my/file.nim"
   assert splitPath(myFile) == (head: "/path/to/my", tail: "file.nim")
-  assert parentDir(myFile) == "/path/to/my"
+  assert parentDir(myFile) == "/path/to/my".unixToNativePath
   assert splitFile(myFile) == (dir: "/path/to/my", name: "file", ext: ".nim")
   assert myFile.changeFileExt("c") == "/path/to/my/file.c"
```
Comment (Araq) [inline lib/pure/os.nim:20]: But now it's confusing that the other examples do not use `.unixToNativePath`. And examples must be simple. These are not unit tests here, they are examples that we ensure keep working.

**Comment by Araq** [comment]
Old solution was preferable, don't teach users `unixToNativePath`, it's not good advice.

File: lib/pure/os.nim (hunk 2 above)
Comment (ringabout) [reply lib/pure/os.nim:20]: > These are not unit tests here, they are examples that we ensure keep working.

The example is broken on windows. This is what I got copying it without changing
```
Hint: C:\Users\blue\nimcache\test5_d\test5_6829A63D57902939FC6B3D487052AFE2BD950EA3.exe  [Exec]
C:\Users\blue\Documents\GitHub\Nim\test5.nim(5) test5
C:\Users\blue\.choosenim\toolchains\nim-#devel\lib\system\assertions.nim(39) failedAssertImpl
C:\Users\blue\.choosenim\toolchains\nim-#devel\lib\system\assertions.nim(29) raiseAssert
C:\Users\blue\.choosenim\toolchains\nim-#devel\lib\system\fatal.nim(53) sysFatal
Error: unhandled exception: C:\Users\blue\Documents\GitHub\Nim\test5.nim(5, 8) `parentDir(myFile) == "/path/to/my"`  [AssertionDefect]
Error: execution of an external program failed: 'C:\Users\blue\nimcache\test5_d\test5_6829A63D57902939FC6B3D487052AFE2BD950EA3.exe '  
```

When users on windows copy the example, running it, finding it not work, they will be more surprised. 
How about 
- removing it 
- or using
```nim
  when defined(posix):
    assert parentDir(myFile) == "/path/to/my"
```
- At least we need to document the example is platform-specific. It only works on Posix.

**Comment by timotheecour** [comment]
> they are examples that we ensure keep working.
> Old solution was preferable

these 2 statements are contradictory; -r:off doesn't run (and old example is broken on windows)

I don't understand why this PR was closed, `unixToNativePath` is an exposed API and is already used in other runnableExamples in std/os; there's nothing wrong with it for writing tests

**Comment by ringabout** [comment]
I'm reopening this PR and fixing the content of the PR(And the broken example on windows).

File: lib/pure/os.nim [hunk 3]
```diff
@@ -11,10 +11,10 @@
 ## retrieving environment variables, reading command line arguments,
 ## working with directories, running shell commands, etc.
 
-runnableExamples("-r:off"):
+runnableExamples:
   let myFile = "/path/to/my/file.nim"
   assert splitPath(myFile) == (head: "/path/to/my", tail: "file.nim")
-  assert parentDir(myFile) == "/path/to/my"
+  assert parentDir(myFile) == "/path/to/my".unixToNativePath
```
Comment (ringabout) [inline lib/pure/os.nim]:
```suggestion
  when defined(posix):
    assert parentDir(myFile) == "/path/to/my"
```

**Comment by ringabout** [comment]
`when defined(posix):` is already used among the `os.nim`.

**Review by timotheecour — APPROVED** [review]

File: lib/pure/os.nim (hunk 2 above)
Comment (Araq) [reply lib/pure/os.nim:20]: `when defined(posix):` is ok, but still a bit weird for example code...

**Pull request merged**